xline/src/storage/index.rs

```diff
@@ -287,6 +290,58 @@ impl IndexOperate for Index {
             .or_insert_with(Vec::new)
             .push(new_rev);
     }
+
+    fn compact(&self, at_rev: i64) -> Vec<KeyRevision> {
+        let mut revs = Vec::new();
+        let mut del_keys = Vec::new();
+
+        let mut inner = self.inner.lock();
+        inner.index.iter_mut().for_each(|(key, revisions)| {
+            if let Some(revision) = revisions.first() {
+                if revision.mod_revision < at_rev {
+                    match revisions.binary_search_by(|rev| rev.mod_revision.cmp(&at_rev)) {
```

Comment:
I think we cound use `partition_point` instead to simplifiy the code, that we get the desired index directly.

Reply:
Done